### PR TITLE
🧹use slice filter to list frameworks

### PR DESCRIPTION
**Pull request by slntopp**

### CODE REVIEW

File: upstream/framework.gql.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ type UpstreamFramework struct {
 	State mondoogql.ComplianceFrameworkState
 }
 
-func ListFrameworks(ctx context.Context, c *gql.MondooClient, scopeMrn string, state *mondoogql.ComplianceFrameworkState) ([]*UpstreamFramework, error) {
+func ListFrameworks(ctx context.Context, c *gql.MondooClient, scopeMrn string, states *[]mondoogql.ComplianceFrameworkState) ([]*UpstreamFramework, error) {
```
Comment (preslavgerchev) [inline upstream/framework.gql.go]: this signature looks a bit weird, why not just pass a slice instead of a ptr to a slice?

Comment (slntopp) [reply upstream/framework.gql.go]: Just what mondoogql expects. afaik if we make slice argument mandatory, then you'd have to send empty slice from the frontend to meet the schema

Comment (preslavgerchev) [reply upstream/framework.gql.go]: Can we make this an implementation detail then? We can still pass the slice to the actual GQL call, but i think the arg in this function should be a `states []mondoogql...`.

Comment (slntopp) [reply upstream/framework.gql.go]: Updated the implementation

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**